src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -399,17 +399,33 @@ public void validate(FailureCollector collector) {
 
     //This method validated the pattern of CMEK Key resource ID.
     private void validateCmekKey(FailureCollector failureCollector) {
-      String[] part = cmekKey.split("/");
-      if ((part.length != 8) || !part[0].equals("projects") || !part[2].equals("locations")
-        || !part[4].equals("keyRings") || !part[6].equals("cryptoKeys")) {
-        failureCollector.addFailure("Invalid cmek key resource Id format. It should be of the form " +
-                                      "projects/<gcp-project-id>/locations/<key-location>/keyRings/" +
-                                      "<key-ring-name>/cryptoKeys/<key-name>", null)
-          .withConfigProperty(NAME_CMEK_KEY);
-      } else if (!part[3].equals(location)) {
-        failureCollector.addFailure(String.format("Cloud KMS region '%s' not available for use " +
-                                                    "with GCS region '%s'.", part[3], location), null)
-          .withConfigProperty(NAME_CMEK_KEY);
+      try {
+        CryptoKeyName cmekKeyName = CryptoKeyName.parse(cmekKey);
+        Boolean isServiceAccountFilePath = isServiceAccountFilePath();
```

Comment:
it's not good that this returns a Boolean and not a boolean. Taking a quick look at the usages, there are possible NullPointerExceptions because of this.

Can you change isServiceAccountFilePath() and isServiceAccountJson() to return booleans? If no value is given, serviceAccountFilePath should be true and serviceAccountJson should be false.

Reply:
Its already a boolean and returns true if no value is given by default. I followed the way, service account was loaded in prepareRun().

Reply:
On develop (https://github.com/data-integrations/google-cloud/blob/07800a568ec0ea9a3c4a06fbfdaee22226e7451e/src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java#L110) it returns a Boolean that can return null. Is it different on another branch or something?